### PR TITLE
fix: key encode use little-endian

### PR DESCRIPTION
**Pull request by happy-v587**
## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive doc describing key layout and on-disk encoding for all data types.

* **New Features**
  * Added a binary score→member key format for ZSets with encoding/decoding and seek support, exposed via public types/APIs.

* **Refactor**
  * Renamed and reorganized internal formatting modules and public exports for consistent naming.

* **Bug Fixes**
  * Corrected byte-ordering and parsing for version/score fields and hardened score handling.

* **Tests**
  * Expanded unit tests covering ordering and many edge cases (NaN, infinities, ±0, boundaries).

[Review Change Stack](https://app.coderabbit.ai/change-stack/arana-db/kiwi/pull/262?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/storage/src/format_member_data_key.rs [hunk 1]
```diff
@@ -61,7 +61,7 @@ impl MemberDataKey {
 
         dst.put_slice(&self.reserve1);
         encode_user_key(&self.key, &mut dst)?;
-        dst.put_u64(self.version);
+        dst.put_u64_le(self.version);
```
Comment (AlexStocks) [inline src/storage/src/format_member_data_key.rs:64]: [P0] 这里把 member key 的版本字段从 big-endian 改成了 little-endian，并且下面的解析也同步切到了 `from_le_bytes`。这会直接改变 Hash/Set/ZSet member data 的落盘 key 格式，但 PR 里没有任何兼容旧数据或迁移逻辑。已有数据库里的 member key 还是按旧格式编码的，升级后二进制会用新格式去构造查询前缀和精确 key，结果就是旧数据会被当成“不存在”。这不是测试缺口，而是一次静默的存储格式升级。至少需要补一个打开旧库的回归测试，并在代码里提供兼容读取或显式迁移路径。

File: src/storage/src/custom_comparator.rs [hunk 2]
```diff
@@ -223,4 +224,351 @@ mod tests {
...
+        let a = ZSetsScoreKey::new(b"key", 1, 0.0, b"member")
+            .encode()
+            .unwrap();
+        let b = ZSetsScoreKey::new(b"key", 1, -0.0, b"member")
+            .encode()
+            .unwrap();
+
+        // 注意：0.0 和 -0.0 的 to_bits() 不同，所以编码不同
+        // 但在比较时，f64::partial_cmp 会认为它们相等
+        assert_eq!(zsets_score_key_compare(&a, &b), Ordering::Equal);
```
Comment (AlexStocks) [inline src/storage/src/custom_comparator.rs:384]: [P0] 这个断言会把一个 RocksDB comparator 契约问题固定下来。`0.0` 和 `-0.0` 的字节表示不同，这里对应的两个 key 也确实不同，但 `zsets_score_key_compare()` 现在会返回 `Ordering::Equal`。对自定义 comparator 来说，两个不同 key 不能比较为相等，否则迭代、覆盖和 compaction 都可能出现未定义行为。NaN payload 也有同样的问题。要么在编码时先把 `-0.0`/NaN 归一化成唯一 bit pattern，要么在数值比较结果为相等时继续用原始 score bits 作为 tie-breaker，不能直接返回 `Equal`。

File: src/storage/src/format_member_data_key.rs (hunk 1 above)
Comment (happy-v587) [reply src/storage/src/format_member_data_key.rs:64]: 不需要兼容

**Comment by ruojieranyishen** [comment]
@happy-v587 请处理下codereview意见

File: src/storage/src/redis_zsets.rs [hunk 3]
```diff
@@ -32,8 +32,17 @@ use snafu::OptionExt;
 use snafu::ResultExt;
 use std::collections::HashSet;
 
-use crate::member_data_key_format::MemberDataKey;
-use crate::zset_score_key_format::{ParsedZSetsScoreKey, ScoreMember, ZSetsScoreKey};
+use crate::format_member_data_key::MemberDataKey;
+use crate::format_zset_score_key::{ParsedZSetsScoreKey, ScoreMember, ZSetsScoreKey};
+
+fn build_zscan_prefix(key: &[u8], version: u64) -> Vec<u8> {
```
Comment (AlexStocks) [inline src/storage/src/redis_zsets.rs]: [P1] `build_zscan_prefix` 构造的 seek key 不是 `ZSetsScoreKey::encode()` 的真实前缀：真实 score key 前面有 8 字节 reserve，user key 也必须经过 `encode_user_key` 并带 `\x00\x00` delimiter；这里直接拼 raw key、一个 `0`、version、再一个 `0`。随后 `IteratorMode::From(&start_key, Forward)` 会从错误的位置 seek，尤其是 key 中包含 `\0` 或比较器按 score-key 格式解析 seek target 时，`ZSCAN` 可能直接返回空结果或跳过本 zset。建议直接用 `ZSetsScoreKey::new(key, version, f64::NEG_INFINITY, &[]).encode_seek_key()?` 作为 cursor=0 的起点，或让 prefix 构造复用 `encode_user_key` 并包含 reserve。

**Review by AlexStocks — COMMENTED** [review]
Read-only review: 发现 1 个需要处理的问题。静态检查 cargo clippy 未完成，原因是本机 Windows GNU 工具链缺少 dlltool.exe。
